index.html

```diff
@@ -8324,41 +8324,308 @@ <h3>Graph Containers</h3>
+      "name": "John Lennon",
+      "born": "1940-10-09",
+      "spouse": "http://dbpedia.org/resource/Cynthia_Lennon"
+    }
+    ****</script>****
+    -->
+    </pre>
+    <pre class="expanded"
+         data-transform="updateExample"
+         data-result-for="Embedding JSON-LD in HTML-original">
+    <!--
+    [{
+      "@id": "http://dbpedia.org/resource/John_Lennon",
+      "http://xmlns.com/foaf/0.1/name": [{"@value": "John Lennon"}],
+      "http://schema.org/birthDate": [
+        {"@value": "1940-10-09", "@type": "http://www.w3.org/2001/XMLSchema#date"}
+      ],
+      "http://schema.org/spouse": [
+        {"@id": "http://dbpedia.org/resource/Cynthia_Lennon"}
+      ]
+    }]
+    -->
+    </pre>
+    <table class="statements"
+           data-result-for="Embedding JSON-LD in HTML-expanded"
+           data-to-rdf>
+      <thead><tr><th>Subject</th><th>Property</th><th>Value</th><th>Value Type</th></tr></thead>
+      <tbody>
+        <tr><td>http://dbpedia.org/resource/John_Lennon</td><td>foaf:name</td><td>John Lennon</td><td>&nbsp;</td></tr>
+        <tr><td>http://dbpedia.org/resource/John_Lennon</td><td>schema:birthDate</td><td>1940-10-09</td><td>xsd:date</td></tr>
+        <tr><td>http://dbpedia.org/resource/John_Lennon</td><td>schema:spouse</td><td>http://dbpedia.org/resource/Cynthia_Lennon</td><td>&nbsp;</td></tr>
+      </tbody>
+    </table>
+    <pre class="turtle"
+         data-content-type="text/turtle"
+         data-transform="updateExample"
+         data-result-for="Embedding JSON-LD in HTML-expanded"
+         data-to-rdf>
+    <!--
+    @prefix foaf: <http://xmlns.com/foaf/0.1/> .
+    @prefix schema: <http://schema.org/> .
+    @prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+    <http://dbpedia.org/resource/John_Lennon> foaf:name "John Lennon";
+       schema:birthDate "1940-10-09"^^xsd:date;
+       schema:spouse <http://dbpedia.org/resource/Cynthia_Lennon> .
+    -->
+    </pre>
+  </aside>
 
   <p>Depending on how the HTML document is served, certain strings may need
-    to be escaped.</p>
+    to be escaped. See
+    <a href="https://www.w3.org/TR/html52/semantics-scripting.html#restrictions-for-contents-of-script-elements">
+      Restrictions for contents of <code>script</code> elements
+    </a> in [[!HTML52]] for details. In particular, the content MAY be enclosed
+    in the HTML comment open (<code>&lt;!--</code>) and comment close (<code>--&gt;</code>) text sequences</p>
+
+   <aside class="example ds-selector-tabs"
+           title="Embedding JSON-LD in HTML with comments">
+    <div class="selectors">
+      <button class="selected" data-selects="original">Original</button>
+      <button data-selects="expanded">Expanded</button>
+      <button data-selects="statements">Statements</button>
+      <button data-selects="turtle">Turtle</button>
+    </div>
+    <pre class="original selected" data-transform="updateExample"
+         data-content-type="text/html">
+    <!--
```

Comment:
I think we'll need an example to more clearly justify this requirement. From what I've read in HTML5.2 and played with so far, it only seems to effect the parsing of the contents of a JavaScript `<script>` tag (either no `type` or one of the explicit [JavaScript MIME Types](https://www.w3.org/TR/html52/semantics-scripting.html#javascript-mime-type)).

For example, changing the [broken `<script>` example in the HTML5.2 spec](https://www.w3.org/TR/html52/semantics-scripting.html#example-8c8534f7) to use a data block, the problem goes away:
```js
<script type="application/ld+json">
  var example = 'Consider this string: <!-- <script>';
  console.log(example);
</script>
<!-- despite appearances, this is actually part of the script still! -->
<script>
  ... // this is the same script block still...
</script>
```

Consequently, I'm not sure that `<!-- -->` is ever necessary for data-blocks.

Reply:
So I'm working on building a [cheerio](https://cheerio.js.org/)-based data-block extractor, and I indeed did make some JSON-LD that can cause this issue. The `<!-- <script>` combo (at least) does cause it to be "overly greedy" and continue to parse beyond the intended closing tag.

I'll try and narrow in on a good/simple example, but it looks like `<!--` and `-->` handling is here to stay...

Reply:
The text uses MAY, and you can certainly avoid using comment blocks, but that requires carefully escaping the content. So yes, it is not truly necessary to do this, but it is a common practice (which we, ourselves, use). The point is that if the content is commented, these need to be considered when processing. (I've left out CDATA, although it could be considered in an XHTML case).

The text says:

> In particular, the content MAY be enclosed in the HTML comment open (`<!--`) and comment close (`-->`) text sequences

We don't require that comments be used, but allow them to be used and place requirements on processors to consider them.

Reply:
It's the "requirements on processors to consider them" that I'm "feeling" right now. 😄 Without that requirement (on implementers), you get to throw the results into `JSON.parse()` and hope for the best. With that requirement, the implementer has to do a good bit of extra effort to extract the intended contents of the `<script>`, and if not done properly essentially creates the reverse scenarios where one might extract only part of the intended data-block.

But authors come before implementers...so... 😖

Reply:
We may need to specify steps for uncommenting and unescaping, but this seems like something we should be able to reference, and I would expect HTML libraries to handle this, but perhaps not.

Reply:
HTML processors are not supposed to process any content within a [data block](https://www.w3.org/TR/html5/semantics-scripting.html#data-block):
> Setting the attribute to any other value means that the script is a data block, which is not processed. None of the script attributes (except type itself) have any effect on data blocks. Authors must use a valid MIME type that is not a JavaScript MIME type to denote data blocks.

Consequently, handling comments in the content would be up to the JSON-LD parser's processing of whatever's returned from `document.querySelector('script[type="application/ld+json"').textContent` or its corollary in a non-DOM-based parser/processor.

If it's something we expect JSON-LD processors to handle directly as input, then we'll need to define it. Otherwise, developers will have to parse out the surrounding comment markup themselves and *then* pass the results to a JSON-LD processor.